docs/design/kademlia-removal.md

```diff
@@ -0,0 +1,86 @@
+# Kademlia Removal
+
+## Abstract
+
+This design document outlines the communication protocol between satellites and 
+storage nodes, network refreshes, and kademlia removal given the satellite opt-in 
+capability for storage nodes. 
+
+## Background
+
+User research on how Storage Node Operators would connect to satellites resulted in a 
+simple solution called Opt-in SNO select. Storage Nodes select which Satellites to work 
+with, hence the Storage Nodes do not need to discover Satellites on the network and 
+Satellites have no benefit from finding Storage Nodes on the network.
+
+At the time of writing, a storage node operator can input a list of trusted satellite 
+IDs and addresses into their configuration file on setup. There are several tardigrade-level 
+satellites already listed as a preset. This list is imported into a Go map data structure 
+within the “trust” package. Future improvements will enable users to manage 
+this list through a web console. Due to the nature of this change, the Storj network 
+at this point in time no longer requires the Kademlia DHT or related entities. 
+
+We will replace this protocol with direct communication between satellites and 
+storage nodes, and keep the network fresh without kademlia node discovery and random 
+lookups.
+
+## Design
+
+### Storage Nodes reach out to satellites that they want to work with
+    - Eventually during the design and implementation of trusted satellite management, 
+    this configured list may be imported into a on-disk data store, eg SQL table. Users will 
+    be able to modify this list through the storage node operator console. However, this is a 
+    separate discussion from this design document. This design is satisfied with a static trusted 
+    satellites list.
+    - Nodes will be able to communicate with satellites through the “transport” and “trust” packages 
+    directly rather than using kademlia to traverse the network to find the address of a given ID.
+
+### Network refreshes at a regular interval
+    - Nodes will keep themselves up to date in the network by pinging all the satellites in their 
+    trusted list every hour. 
+    - Satellites will ping the nodes back
```

Comment:
Can we specify that this is rate limited.  Like if we haven't heard from them in X amount of time?

Reply:
If the satellites haven't heard from the nodes in x amount of time?

Reply:
satellites should ping the node back if the node has asked the satellite to ping it, however often the node asks

Reply:
> however often the node asks

Really? Why? If a satellite has a set policy that uptime checks are once/hour, it's reasonable that if a node keeps asking for uptime pings over and over again, you can ignore them. And potentially even DQ the node.

I suggested above that the satellite can tell the node its policy for how often the node should check in. Based on that, if a node is asking for ping-backs too frequently, I could see that as something to protect against.

Reply:
okay, it's not a hard requirement that we ping back every time the node asks, but i also don't think it's an "attack" worth defending against. the storage node operator has to pay the same resources the satellite does. it would be really hard to DoS a satellite this way. a misconfigured node could be annoying though, i agree.